From 2a848025a9d6f499389d2decc9fccffbdd33b4e5 Mon Sep 17 00:00:00 2001
From: Roger Leigh <rleigh@debian.org>
Date: Tue, 7 Dec 2010 23:25:01 +0000
Subject: [PATCH] dh: Don't treat build-arch and build-indep sequences specially

Currently, the build-arch sequence adds "-a", and build-indep adds
"-i" to the debhelper options.  This makes sense for binary and
install targets.  However, build targets are different; we don't
run any debhelper packaging commands at this point in the build where
this would make sense.  If separating architecture dependent and
independent parts of the build, we will be writing custom overrides
and/or rules in debian/rules.  If not, then the default actions will
suffice.

This change makes the build-arch and build-indep sequences behave
identically.  As a result, their behaviour is only different when
specifically requested by the user.  This means the behaviour is
identical to previous dh releases by default.  "dh build" will
run both the build-arch and build-indep sequences, but the sequence
actions will only be run the first time through.
---
 dh |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/dh b/dh
index 9031747..24db303 100755
--- a/dh
+++ b/dh
@@ -621,8 +621,7 @@ my @packages=@{$dh{DOPACKAGES}};
 # Get the options to pass to commands in the sequence.
 # Filter out options intended only for this program.
 my @options;
-if ($sequence eq 'build-arch' ||
-    $sequence eq 'install-arch' ||
+if ($sequence eq 'install-arch' ||
     $sequence eq 'binary-arch') {
 	push @options, "-a";
 	# as an optimisation, remove from the list any packages
@@ -630,8 +629,7 @@ if ($sequence eq 'build-arch' ||
 	my %arch_packages = map { $_ => 1 } getpackages("arch");
 	@packages = grep { $arch_packages{$_} } @packages;
 }
-elsif ($sequence eq 'build-indep' ||
-       $sequence eq 'install-indep' ||
+elsif ($sequence eq 'install-indep' ||
        $sequence eq 'binary-indep') {
 	push @options, "-i";
 	# ditto optimisation for arch indep
-- 
1.7.2.3

